app/src/test/java/de/xikolo/testing/unit/RecentCoursesTest.kt

```diff
@@ -0,0 +1,79 @@
+package de.xikolo.testing.unit
+
+import de.xikolo.controllers.helper.MAX_SHORTCUTS
+import de.xikolo.storages.RecentCourse
+import de.xikolo.storages.RecentCoursesStorage
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class RecentCoursesTest {
+
+    @Test
+    fun testAddNewCourseToEmpty() {
+        var set = LinkedHashSet<RecentCourse>()
+        val testSet = LinkedHashSet<RecentCourse>()
+        assertTrue(set.size == 0)
+
+        set = RecentCoursesStorage.addCourseToRecentCourses("id1", "Course1", set)
+        assertTrue(set.size == 1)
+        testSet.add(Pair("id1", "Course1"))
+        assertTrue(set == testSet)
+    }
+
+    @Test
+    fun testAddNewCourseToPartlyFilled() {
+        var set = LinkedHashSet<RecentCourse>()
+        val testSet = LinkedHashSet<RecentCourse>()
+        set = RecentCoursesStorage.addCourseToRecentCourses("id1", "Course1", set)
+        testSet.add(Pair("id1", "Course1"))
+
+        set = RecentCoursesStorage.addCourseToRecentCourses("id2", "Course2", set)
+        testSet.add(Pair("id2", "Course2"))
+        assertTrue(set == testSet)
+
+        // check order
+        assertTrue(set.last() == Pair("id2", "Course2"))
```

Comment:
checking order in a set does not make sense

Reply:
I think we used a `LinkedHashSet` on purpose, to have unique entries and defined order.